tests/Unit/IO/Test_H5.py

```diff
@@ -0,0 +1,89 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+from spectre import DataStructures
+from spectre.IO.H5 import IOH5File as  spectre_h5
+import unittest
+import numpy as np
+import os
+
+
+class TestIOH5File(unittest.TestCase):
+    # Test Fixtures
+    def setUp(self):
+        self.file_name = "./pythontest.h5"
+        self.data_1 = [1.0, 3.5]
+        self.data_1_array = np.array(self.data_1)
+        self.data_2 = [3.0, 10.3]
+        self.data_2_array = np.array(self.data_2)
+        self.cd = os.getcwd()
+        if "pythontest.h5" in os.listdir(self.cd):
+            os.remove(self.file_name)
+
+    # Test whether an H5 file is created correctly, delete a file with the name
+    # 'pythontest.h5' if it already exists.
+    def test_name(self):
+        file_spec = spectre_h5.H5File(self.file_name, 1)
+        self.assertEqual(self.file_name, file_spec.name())
+        file_spec.close()
+        os.remove(self.file_name)
+
+    # Test whether a dat file can be added correctly
+    def test_insert_dat(self):
+        file_spec = spectre_h5.H5File(self.file_name, 1)
+        file_spec.insert_dat("/element_data", ["Time", "Value"], 0)
+        datfile = file_spec.get_dat("/element_data")
+        self.assertEqual(datfile.get_version(), 0)
+        file_spec.close()
+        os.remove(self.file_name)
+
+    # Test whether data can be added to the dat file correctly
+    def test_append(self):
+        file_spec = spectre_h5.H5File(self.file_name, 1)
+        file_spec.insert_dat("/element_data", ["Time", "Value"], 0)
+        datfile = file_spec.get_dat("/element_data")
+        datfile.append(self.data_1)
```

Comment:
Can you call `append` with the Numpy array instead of the list?

Reply:
This is a pretty complicated process because it would require a `from_numpy` function from `PyObject`'s to `std::vector`'s , also it seems to me that the python `list` is more analogous to `std::vector` than numpy `array` is.

Reply:
As discussed with @nilsdeppe earlier today, for the `append` function to be usable in practice in Python it is quite important that it can take either a list, a numpy array, or possibly other iterable types. If we don't need that functionality now and it's too hard to add now, please open an issue and/or add a comment to the Python help string for this function.

Reply:
I've added an issue, #1513 listing the points that I could remember we would like to change/improve once we have some experience with using the wrappers. Please feel free to edit my comment to add more points (or add a comment to the issue and I'll edit my comment so we have the list in one place)